kernel/sched.c

```diff
@@ -1258,7 +1262,7 @@ static int32_t z_tick_sleep(int32_t ticks)
 	timeout = Z_TIMEOUT_TICKS(ticks);
 #else
 	ticks += _TICK_ALIGN;
-	timeout = (k_ticks_t) ticks;
+	timeout = ticks;
```

Comment:
Should this also be `Z_TIMEOUT_TICKS` rather than casting the int to a struct

Reply:
yep, I missed that we had it to legacy api, just changing it. Thanks for pointing it out.

Reply:
The initial change was correct. There was no casting to structure, because modified line was executed when `CONFIG_LEGACY_TIMEOUT_API=y`.

Fix in #30842.